react-game/react-game/src/Components/Cards.js

```diff
@@ -0,0 +1,176 @@
+import React from 'react';
+
+export class Cards extends React.Component {
+ constructor(props) {
+  super(props);
+  this.state = JSON.parse(localStorage.getItem('state_cards')) ? JSON.parse(localStorage.getItem('state_cards')) : {array: this.props.cardsarray, cards1: [], cards2: [], cards3: [], cards4: [], trash: [], style1: '0.5rem #00ff4c solid', style2: '0.5rem #00ff4c solid', style3: '0.5rem #00ff4c solid', style4: '0.5rem #00ff4c solid', cont1: 0, cont2: 0,cont3: 0,cont4: 0,a1:false, pressKey: false, volume: this.props.volume()/100}
+   this.setState({style1: '0.5rem #00ff4c solid',style2: '0.5rem #00ff4c solid',style3: '0.5rem #00ff4c solid',style4: '0.5rem #00ff4c solid'});
+   setInterval(() => {
+      this.checkNewGame.call(this);
+   }, 1000);
+}
+
+_handleKeyDown = (event) => {
+   if (!event.repeat) {
+      switch( event.key) {
+         case '1':
+            this.deleteCard.call(this, {target:{id:'cards1'}});
+              break;
+         case '2':
+            this.deleteCard.call(this, {target:{id:'cards2'}});
+              break;
+          case '3':
+            this.deleteCard.call(this, {target:{id:'cards3'}});
+              break;
+         case '4':
+            this.deleteCard.call(this, {target:{id:'cards4'}});
+              break;
+          case ' ':
+            this.deleteCard.call(this, {target:{id:'trash'}});
+              break;
+          default:
+              break;
+      }
+   }
+}
+
+componentDidMount(){
+   document.addEventListener("keyup", this._handleKeyDown);
+}
+
+componentWillUnmount() {
+   document.removeEventListener("keyup", this._handleKeyDown);
+}
+
+ checkNewGame() {
+    if (this.props.newgame) {this.setState({array: this.props.cardsarray, cards1: [], cards2: [], cards3: [], cards4: [], trash: [], style1: '0.5rem #00ff4c solid', style2: '0.5rem #00ff4c solid', style3: '0.5rem #00ff4c solid', style4: '0.5rem #00ff4c solid', cont1: 0, cont2: 0,cont3: 0,cont4: 0})
+    localStorage.setItem('state_cards',JSON.stringify(this.state));
+   }
+ }
+
+ sound(path){
+    this.setState({volume: this.props.volume() /100});
+    this.audio = new Audio(path);
+    this.audio.volume = this.state.volume;

```

Comment:
if you want to be sure that you assign updated state here you should make it in callback of the setState, which you execute in 52 row